### PR TITLE
Pick save table auto-detect extension

### LINKED ISSUE #479: Feature Request: Allow Pick to save out as ASCII table
Someone asked me why `Pick` only allows saving out to FITS table, not ASCII. The use case is that user wants to use the output with `tweakreg` (part of [DrizzlePac](http://drizzlepac.stsci.edu/)), which only accepts ASCII table.

### PR DESCRIPTION
**Pull request by pllim**
Fix #479 

p.s. The "comment" is only saved out as FITS header keyword but not as a comment line in ASCII. See astropy/astropy#6079

### CODE REVIEW

**Comment by ejeschke** [comment]
@pllim, do you know what minimum astropy version supports the change in this PR?

**Comment by pllim** [comment]
It seems like you already use "overwrite" keyword. I think that's 1.2 or 1.3 (hard to check on phone). So the changes here are compatible with Astropy that you already use.

**Comment by pllim** [comment]
I notice that your Astropy requirement is 0.3. That is a very old version. Is it okay to bump the version up to 1.x?

**Comment by ejeschke** [comment]
> It seems like you already use "overwrite" keyword. I think that's 1.2 or 1.3 (hard to check on phone). So the changes here are compatible with Astropy that you already use.

I just tested with python3 and astropy 1.1.1.  `IORegistryError` doesn't seem to be defined in that version at least, but the `overwrite` keyword is supported.

Could we just do something like:
```python
format = None
if filepath.lower().endswith('.txt'):
    format = 'ascii.commented_header'
tbl.write(filepath, overwrite=True, format=format)
```
?

**Comment by pllim** [comment]
@ejeschke , okay, I implemented your suggestion. Also fixed the syntax as addressed by astropy/astropy#6097 and bumping Astropy requirement to v1.x.

File: setup.py [hunk 1]
```diff
@@ -105,7 +105,7 @@
 # https://groups.google.com/forum/#!topic/astropy-dev/urYO8ckB2uM
 
 # pretty much needed
-install_requires = ['astropy>=0.3', 'numpy>=1.7', 'qtpy>=1.1',
```
Comment (ejeschke) [inline setup.py]: @pllim, can we treat this change as a separate PR?  In general, I am not opposed, but there are some considerations which are unrelated to this PR and I'd rather treat it as a separate issue.

Comment (pllim) [inline setup.py]: Okay. Removed.

**Comment by pllim** [comment]
Removed Astropy bump. Please review.

**Comment by ejeschke** [comment]
Looks good.

**Pull request merged**